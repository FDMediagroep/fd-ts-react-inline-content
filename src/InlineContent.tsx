import {BulletPoints, BulletPointsStyle} from './components/BulletPoints';
import {LinkBlock, LinkBlockStyle} from './components/LinkBlock';
import {NumberFrame, NumberFrameStyle} from './components/NumberFrame';
import {Quote, QuoteStyle} from './components/Quote';
import {StackFrame, StackFrameStyle, StackFrameDataType} from './components/StackFrame';
import {Stock, StockStyle} from './components/Stock';
import {Summary, SummaryStyle} from './components/Summary';
import {TextFrame, TextFrameStyle} from './components/TextFrame';

export {BulletPoints, BulletPointsStyle};
export {LinkBlock, LinkBlockStyle};
export {NumberFrame, NumberFrameStyle};
export {Quote, QuoteStyle};
export {StackFrame, StackFrameStyle, StackFrameDataType};
export {Stock, StockStyle};
export {Summary, SummaryStyle};
export {TextFrame, TextFrameStyle};
