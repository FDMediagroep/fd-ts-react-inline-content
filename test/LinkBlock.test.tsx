import React from 'react';
import TestRenderer from 'react-test-renderer';
import { LinkBlock } from '../src/InlineContent';

describe('LinkBlock', ()  => {
    const title = "Lees ook";
    const description = "Curabitur quis luctus est, vitae sodales enim";
    const url = "https://fd.nl/ondernemen/1254849/douane-20-tot-30-meer-aangiftes-door-groei-in-webwinkelen";

    it('should render align right correctly', () => {
        let component = TestRenderer.create(<LinkBlock {...{title, description, url}} />);
        expect(component.toJSON()).toMatchSnapshot();

        component = TestRenderer.create(<LinkBlock {...{title, description, url}} align='right'/>);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should render align block correctly', () => {
        let component = TestRenderer.create(<LinkBlock {...{title, description, url}} align='block'/>);
        expect(component.toJSON()).toMatchSnapshot();
    });

});
