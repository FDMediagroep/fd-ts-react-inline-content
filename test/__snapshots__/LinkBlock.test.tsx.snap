// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkBlock should render align block correctly 1`] = `
<a
  className="inline-content fd-card article related-link block"
  href="https://fd.nl/ondernemen/1254849/douane-20-tot-30-meer-aangiftes-door-groei-in-webwinkelen"
  target="_blank"
>
  <p
    className="fd-typography article-p-inline-link-block"
    textStyle="article-p-inline-link-block"
  >
    Lees ook
  </p>
  <h2
    className="fd-typography article-h-inline-link-block"
    textStyle="article-h-inline-link-block"
  >
    Curabitur quis luctus est, vitae sodales enim
  </h2>
  <i
    className="icon-chevron-right"
  />
</a>
`;

exports[`LinkBlock should render align right correctly 1`] = `
<a
  className="inline-content fd-card article related-link right"
  href="https://fd.nl/ondernemen/1254849/douane-20-tot-30-meer-aangiftes-door-groei-in-webwinkelen"
  target="_blank"
>
  <p
    className="fd-typography article-p-inline-link-block"
    textStyle="article-p-inline-link-block"
  >
    Lees ook
  </p>
  <h2
    className="fd-typography article-h-inline-link-block"
    textStyle="article-h-inline-link-block"
  >
    Curabitur quis luctus est, vitae sodales enim
  </h2>
  <i
    className="icon-chevron-right"
  />
</a>
`;

exports[`LinkBlock should render align right correctly 2`] = `
<a
  className="inline-content fd-card article related-link right"
  href="https://fd.nl/ondernemen/1254849/douane-20-tot-30-meer-aangiftes-door-groei-in-webwinkelen"
  target="_blank"
>
  <p
    className="fd-typography article-p-inline-link-block"
    textStyle="article-p-inline-link-block"
  >
    Lees ook
  </p>
  <h2
    className="fd-typography article-h-inline-link-block"
    textStyle="article-h-inline-link-block"
  >
    Curabitur quis luctus est, vitae sodales enim
  </h2>
  <i
    className="icon-chevron-right"
  />
</a>
`;
